### ISSUE TITLE
Unreachable code warning for platform-specific branches

### ISSUE THREAD

**Issue by cscrimge** [microsoft/pyright#10114]
**Describe the bug**
Platform-specific branches trigger an unreachable code warning

**Sample Code**
```python
import os
import sys

if os.name == "nt":
    print("Windows")  # Code is unreachable

if sys.platform == "win32":
    print("Windows")  # Code is unreachable
```

![Image](https://github.com/user-attachments/assets/a735339f-5999-4df6-b9b7-8720f7aaff78)

**VS Code extension or command-line**
Running pyright 1.1.396 as a Neovim LSP server

**Comment by erictraut** [comment]
Yes, pyright is working as designed here, so this isn't a bug.

First, pyright never emits a warning for unreachable code. It does emit an "unused tag" along with a hint. This tells an editor to display the code in a dimmed manner without flagging it as an error or warning. This is because there are many legit reasons for code to be unreachable. Pyright wants to bring it to your attention but not flag it as an error or warning.

Second, the [Python typing spec](https://typing.python.org/en/latest/spec/directives.html#version-and-platform-checking) indicates that type checkers should statically evaluate certain version and platform checks. Many stubs and libraries rely on this behavior. You can configure which platform you want pyright to assume. By default, it assumes that you're running the code on the same platform as your development environment. If you'd like to override this default, you can specify the `pythonVersion` and `pythonPlatform` in your pyright config. If you'd like to tell pyright to assume that the code runs on all platforms, you can specify `All` for the `pythonPlatform`.
